### PR TITLE
Implement spotify API connection & search

### PR DESCRIPTION
**Pull request by Paattis**
* add class for handling Spotify API connections, add AccessToken model to store access tokens
   * add route for searching songs `GET /songs/search/<searchTerm>`
   * add new config values to readme
   
Solves #3 #4

### CODE REVIEW

File: classes/DatabaseTokenCacheHandler.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      AccessToken.expire_time>datetime.now()+timedelta(seconds=120)
+    ).order_by(desc(AccessToken.id)).first()
+
+    if token:
+      self.access_token = token.access_token
+      self.tokenExpireTime = token.expire_time
+
+    # if a valid token doesn't exist, fetch a new one from the Spotify API
+    if not token:
+      requestCode = base64.b64encode(f'{SPOTIPY_CLIENT_ID}:{SPOTIPY_CLIENT_SECRET}'.encode('utf-8'))
```
Comment (therealnikke2) [inline classes/DatabaseTokenCacheHandler.py]: Extract "no token" situation handling to separate function, eg. "fetchToken()"

File: classes/DatabaseTokenCacheHandler.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      self.access_token = data['access_token']
+      self.tokenExpireTime = datetime.now() + timedelta(seconds=data['expires_in'])
+
+
+      # finally save the token
+      newToken = AccessToken(access_token=self.access_token, expire_time=self.tokenExpireTime)
+      db.session.add(newToken)
+      db.session.commit()
+      print('Returning token', data)
+      return data if as_dict else data['access_token']
```
Comment (therealnikke2) [inline classes/DatabaseTokenCacheHandler.py]: Refactor to single return clause?

Comment (Paattis) [reply classes/DatabaseTokenCacheHandler.py]: 👍

File: classes/DatabaseTokenCacheHandler.py (hunk 1 above)
Comment (Paattis) [reply classes/DatabaseTokenCacheHandler.py]: Good idea 👍

**Review by therealnikke2 — APPROVED** [review]

**Pull request merged**